socorro/lib/task_manager.py

```diff
@@ -39,10 +39,18 @@ def respond_to_SIGTERM(signal_number, frame, logger=None):
     parameters:
         signal_number - unused in this function but required by the api.
         frame - unused in this function but required by the api.
+        target - an instance of a class that has a member called 'task_manager'
+                 that is a derivative of the TaskManager class below.
     """
-    if logger:
-        logger.info('detected SIGTERM')
-    raise KeyboardInterrupt
+    if target:
+        target.config.logger.info('detected SIGTERM')
+        # by setting the quit flag to true, any calls to the 'quit_check'
+        # method that is so liberally passed around in this framework will
+        # result in raising the quit exception.  The current quit exception
+        # is KeyboardInterrupt
+        target.task_manager.quit = True
```

Comment:
If it was `target.task_manager.quit()` instead, it'd make more sense to put the `self.config.logger.info('detected SIGTERM quit')` in there instead.